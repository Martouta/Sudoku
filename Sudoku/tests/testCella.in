3 3

1 

2 
3

1

6

5

6

1

3

2
1

3

5

2
1

1

12 

7

8

9
3

11
2 5

4

12

14

13

0






Explicación:

Escribe el x y el y de la celda
3 3 //Introducimos la posición x e y de la celda, que no pueden ser menores que 0, por lo contrario recibiremos una excepción

1 //Inicialmente, la celda está vacía y por eso al hacer el getNumero obtenemos un -1

2 //Introducimos un nuevo número a esa casilla con el setNumero, en este caso el 3 y se introduce sin ningún problema ya que no está fijada
Nuevo nÃºmero?
3

1 //getNumero de la casilla y esta vez ya tiene valor, así que obtenemos un 3 por el caso anterior

6 //Al comprobar si la celda está vacía, se muestra un booleano que indica que no(false) ya que contiene un 3

5 //Eliminamos el contenido de la celda, y como no está fijada se borra

6 //Esta vez, al comprobar si la celda está vacía, se muestra un booleano que indica que si (true) ya que hemos borrado su contenido anteriormente

1 //Al hacer un get vemos que vuelve a estar vacía y nos muestra un -1 (no tiene valor)

3 //Fijamos la casilla y nos sale la excepción de que necesita tener valor previo antes de ser fijada

2 //introducimos un 1
Nuevo nÃºmero?
1

3 //Esta vez la celda ya puede ser fijada

5 //Al intentar borrar el valor de la celda recibimos una excepción de que esta está fijada, y por eso no se puede borrar su contenido

2 //Como ya está fijada, tampoco se puede insertar ningún valor en ella
Nuevo nÃºmero?
1

1 //Hacemos el getNumero para ver que, aunque esté fijada, se puede obtener igualmente

12 //Nos indica que la celda está fija

7 //Obtenemos su coordenada x

8 //Obtenemos su coordenada y

9 //Al intentar modificar su coordenada x, como ya se había introducido previamente nos salta una excepción, ya que no se puede modificar una coordenada ya introducida

11 //Lo mismo que en el caso anterior, pero esta vez intentamos modificar sus dos coordenadas

4 //Liberamos la celda

12 //En esta ocasión nos indica con un booleano que no está fijada, ya que la hemos liberado

13 //Nos devuelve un booleano (false) ya que no está bloqueada

14 // La bloqueamos

13 //En esta ocasión es un true, ya que está bloqueada

0