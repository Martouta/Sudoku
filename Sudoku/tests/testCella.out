-1

3

false

true

No se puede fijar una casilla vacia

La casilla esta fija y no se puede modificar

La casilla esta fija y no se puede modificar

1

true

3

3

esta casilla ya tiene posicion y no se puede modificar

esta casilla ya tiene posicion y no se puede modificar

false

true

true
